pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -935,6 +935,17 @@ func (s *DiscoveryServer) removeCon(conID string, con *XdsConnection) {
 	}
 }
 
+func (s *DiscoveryServer) workloadLabels(proxy *model.Proxy) model.LabelsCollection {
+	ipAddresses := proxy.IPAddresses
+	labels := make([]model.Labels, 0, len(ipAddresses))
+	for _, ip := range ipAddresses {
+		if workload := s.WorkloadsByID[ip]; workload != nil {
```

Comment:
WorkloadsByID only set in k8s env

Reply:
That was one of my questions: How would proxies identify for non-K8s applications?
Is it required to register MESH_INTERNAL ServiceEntries for these? 
Or use Consul?

`Proxy.Metadata` and the current approach using `Proxy.ServiceInstances` don't work, as the metadata is fix and doesn't follow Pod labels, and ServiceInstances doesn't exist for consumer only applications.
Consumer only non-K8s applications should suffer from the same problem.

Reply:
WorkloadsByID is only used by incremental EDS feature.

Reply:
Which only supported in k8s

Reply:
I understand that.
Maybe it would be possible to store the labels in another location at the DiscoveryServer where other Registries can store and update the labels too.
That shouldn't be too hard actually and could probably also use the IPAddress as key, the Consul Registry already uses this for matching a Proxy to a ServiceInstance.

Reply:
Maybe another approach could be to add a method `GetProxyWorkloadLabels` to `ServiceRegistry`.
On Kubernetes this could still get dynamic updates.
I can have a try for this approach today.